### ISSUE TITLE
The normalize function of LGA

### ISSUE THREAD

**Issue by youmi-zym** [feihuzhang/GANet#29]
Hi,
    In my experiments, I replace the maximum normalization function in LGA with softmax, the loss of final cost volume can't decrease at all, that's kind of unexplainable. And I think maximum normalization works more fierce than softmax, it's not continuous in mathematics. But surprisingly, I have also tried to normalize the cost volume in my self-designed network before, and maximum normalization works better than normalized by variance or sigmoid. I still cannot find out the real things to explain it. Can you share your opinions about this?

**Comment by feihuzhang** [comment]
I don't understand what you mean about "maximum normalization".
You mean L1 normalization or L_infinity ? Currently, I only use is L1 normalization in the experiments. It's a little better than softmax. L_infinity is not stable.

**Comment by youmi-zym** [comment]
The maximum normalization, I mean :
```python
F.normalize(x, p=1, dim=1)
```
In my experiment, the 3rd disparity map loss does not decrease at all from the start of training by replacing softmax with maximum normalization.
Here is the original code in DispAgg module:
```python
x = self.lga(x, lg1)
x = self.softmax(x)
x = self.lga(x, lg2)
x = F.normalize(x, p=1, dim=1)
return self.disparity(x)
```

And here is the version I modified:
```python
x = self.lga(x, lg1)
x = self.softmax(x)
x = self.lga(x, lg2)

# here is what I modified
x = F.softmax(-x, dim=1)

return self.disparity(x)
```

**Comment by feihuzhang** [comment]
You should not use two softmax here. After the first softmax and Lga layers the value of all elements are among (0,1). Softmax doesn't here.
